File: first/java/Daytime.java

public class Daytime extends Extras {

    private final float dayTime = 1.25f;
    private final int factor;

    public Daytime(int extraHours) {
        super(extraHours);
        this.factor = extraHours;
    }

    @Override
    public float extraPay(int valueHour) {
        return (valueHour * dayTime) * factor;
    }
}

File: first/php/library/daytime.php
<?php
class Daytime extends Extras {
    
    private float $dayTime = 1.25;
    private int $factor;

    public function __construct(int $extraHours) {
        parent::__construct($extraHours);
        $this->factor = $extraHours; 
    }

    public function extraPay(int $valueHour): float {
        return ($valueHour * $this->dayTime) * $this->factor;
    }
}
    ?>

